features/src/autogluon/features/generators/text_ngram.py

```diff
@@ -73,6 +84,8 @@ def _transform(self, X: DataFrame) -> DataFrame:
             return DataFrame(index=X.index)
         try:
             X_out = self._generate_ngrams(X=X)
+            if self.prefilter_tokens:
```

Comment:
add `# TODO: Optimize for inference`

This can be optimized heavily to be much faster, but is more complex than necessary for initial implementation.
The idea is to update the vectorizer to only produce the chosen set of ngrams, removing the need for any change at all to the `transform` method.

Reply:
agree. I'll add the comment for now.